using_images/other_images/jenkins.adoc

```diff
@@ -106,6 +106,66 @@ To use a persistent volume store, the cluster administrator must define a link:.
 
 Once selected, you must link:../../dev_guide/templates.html#creating-resources-from-a-template[instantiate] the template to be able to use Jenkins.
 
+== Using Jenkins as a Source-To-Image builder
+
+If you want to customize the official OpenShift Jenkins image, you can use
+Docker layering or you can use the image as a Source-To-Image builder.
+You can use link:../../architecture/core_concepts/builds_and_image_streams.html#source-build[S2I]
+to copy your custom Jenkins Jobs definitions, additional
+plugins or replace the provided `config.xml` file with your own, custom, configuration.
+
+In order to include your modifications in the Jenkins image, you need to have a Git
+repository, with the following directory structure:
+
+* `plugins` folder that contains binary Jenkins plugins you want to copy into Jenkins 
+* `plugins.txt` file that list the plugins you want to install (see the section above)
+* `configuration/jobs` folder that contains the Jenkins job definitions
+* `configuration/config.xml` file that contains your custom Jenkins configuration
+
+Note that the content of `configuration` folder will be copied into the `/var/lib/jenkins`
+folder, so you can also include additional files (like `credentials.xml`, etc.).
+
+For example, you can use following build configuration to customize the Jenkins
+image in OpenShift:
+
+[source,json]
+----
+{
+  "kind": "BuildConfig",
+    "apiVersion": "v1",
+    "metadata": {
+      "name": "custom-jenkins-build"
+    },
+  "spec": {
+    "source": { <1>
+      "type": "Git",
+      "git": {
+        "uri": "git://github.com/custom/repository.git"
+      }
+    },
+    "strategy": { <2>
+      "type": "Source",
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "jenkins-1-centos7:latest"
```

Comment:
I take it for OSE we should conditionalize this to instead just show `jenkins:latest`?

```
$ oc get istag -n openshift | grep jenkins

jenkins:1                                    registry.access.redhat.com/openshift3/jenkins-1-rhel7:latest                          2 weeks ago   3634ee8eec8ac268a86133e1ee560bdaa7453dd566e86c08e2402c82145d8d8f
jenkins:latest                               registry.access.redhat.com/openshift3/jenkins-1-rhel7:latest                          2 weeks ago   3634ee8eec8ac268a86133e1ee560bdaa7453dd566e86c08e2402c82145d8d8f
```